src/main/webapp/scripts/tests/comment-post-test.js

```diff
@@ -0,0 +1,77 @@
+import { postVideoComment } from '/scripts/post-comment.js';
+
+const VIDEO_ID = 'sample video id';
+const COMMENT_TEXT = 'sample comment';
+const REQUEST_BODY = {
+    'path': '/youtube/v3/commentThreads?part=snippet',
+    'method': 'POST',
+    'body': {
+        "snippet": {
+            "videoId": VIDEO_ID,
+            "topLevelComment": {
+                "snippet": {
+                    "textOriginal": COMMENT_TEXT,
+                },
+            },
+        },
+    },
+};
+
+QUnit.module('Post a comment', {
+    before: loadClientAsync,
+    beforeEach: () => { sinon.stub(gapi.client, 'request'); },
+    afterEach: () => { gapi.client.request.restore(); },
+});
+
+QUnit.test('Throw an error', function (assert) {
+    var done = assert.async();
+    const errorMsg = 'error message';
+
+    gapi.client.request.withArgs(REQUEST_BODY).returns(new Promise(() => {
+        throw new CustomError(errorMsg);
+    }));
+
+    postVideoComment(VIDEO_ID, COMMENT_TEXT)
+        .catch(error => {
+            assert.equal(error, errorMsg, 'Pass the error message to the callback function');
+            done();
+        });
+});
+
+QUnit.test('Successful post', function (assert) {
+    var done = assert.async();
+
+    gapi.client.request.withArgs(REQUEST_BODY).returns(new Promise((resolve) => {
```

Comment:
This might look repetitive and odd because of the file is small and too complex but you should create a helper test function that creates the expected request body that takes in video id and comment. You want to do this because you want to check "Given that I call postVideoComment with VIDEO_ID and COMMENT_TEXT, was gapi called with a request body that contains VIDEO_ID and COMMENT_TEXT in the correct fields?" You want this to be clear in the test itself without a reader going back up to check what REQUEST_BODY is.

Reply:
Yap Done!

Reply:
I meant something more like 

```gapi.client.request.withArgs(createExpectedRequestBody(VIDEO_ID, COMMENT_TEXT)).returns(...)```

You can keep your new assertion as it is now but instead of REQUEST_BODY, it could be createExpectedRequestBody(...). Then when you read this unit test as a whole you see that when you call your method with video id and comment text, you successfully called the api with request body that contains the same video id and comment text.

Reply:
Oow my bad, I misunderstood the first time. It has been modified as suggested. I thought this will re-implement the logic from the main code, but I can definitely see your point of how it improves test readability.